### ISSUE TITLE
Support @psalm-taint-escape docblock on closure validation rules

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1351]
## Context

Custom Rule classes already support a class-level `@psalm-taint-escape` annotation, read by `ValidationRuleAnalyzer` when the rule object appears in a rules array ([docs](https://github.com/psalm/psalm-plugin-laravel/blob/master/docs/contributing/taint-analysis.md#custom-rule-classes)). Closure rules contribute no escape today because static analysis cannot see into the closure body. Splitting a one-off check into a dedicated Rule class only to carry the annotation is heavier than the inline closure Laravel encourages.

Prompted by #1335, where the value is made safe by an `is_subclass_of` check inside a closure rule.

## Proposal

Honour a docblock placed directly on the closure literal in a rules array:

```php
$request->validate([
    'dataTable' => ['required', 'string',
        /** @psalm-taint-escape callable */
        function (string $attribute, mixed $value, \Closure $fail) {
            if (! is_subclass_of($value, DataTable::class)) {
                $fail('The selected data table is invalid.');
            }
        },
    ],
]);
```

- Applies in both inline `validate()` arrays and FormRequest `rules()` arrays (both funnel through `extractRulePairsFromArrayNode()`).
- Same trust model as the class-level annotation: a developer assertion, so a wrong annotation becomes a false negative.
- Same tag semantics: bare form only, any `TaintKind` name, conditional form ignored.
- No annotation means no escape, current behavior unchanged.

## Sketch

- New branch in `extractRulePairsFromArrayNode()` for `Closure` / `ArrowFunction` items, reading the doc comment from the closure node or its enclosing array item (attachment position to be verified empirically).
- Reuse the `@psalm-taint-escape` tag parsing from `classRuleRemovedTaints()` via a shared helper.
- Carry the resolved bits through a synthetic rule segment so `resolveRuleSegments()` can OR them into the field's removed-taints mask.

## Out of scope

Magic property reads on plain `Request` (`$request->dataTable`) do not receive inline-validate escapes at all today; that gap is independent of this feature and applies equally to Rule classes.
